CDEFramework/Libraries/LegacyDictionarySupport/GlossaryAPIClient.cs

```diff
@@ -0,0 +1,152 @@
+            }
+
+            // Set up search param string: {dictionary}/{audience}/{language}/{id:long}
+            string[] searchParams = { dictionary, audience, language, id, useFallback.ToString() };
+            string searchParam = string.Format("{0}/{1}/{2}/{3}?useFallback={4}", searchParams);
+
+            //Get the HTTP response content from GET request
+            HttpContent httpContent = ReturnGetRespContent("Terms", searchParam);
+            if (httpContent != null)
+            {
+                Task<GlossaryTerm> term = ReadAsJsonAsync<GlossaryTerm>(httpContent);
+                var resultTerm = term.Result;
+                return resultTerm;
+            }
+            else
+            {
+                return null;
+            }
+            
+        }
+
+        /// <summary>
+        /// Gets the response content of a GET request.
+        /// </summary>
+        /// <param name="path">Path for client address</param>
+        /// <param name="param">Param in URL</param>
+        /// <returns>HTTP response content</returns>
+        public HttpContent ReturnGetRespContent(String path, String param)
+        {
+            HttpResponseMessage response = null;
+            HttpContent content = null;
+            String notFound = "NotFound";
+
+            
+            _client.DefaultRequestHeaders.Accept.Clear();
+            _client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+
+            // We want this to be synchronus, so call Result right away.
+            //NOTE: When using HttpClient.BaseAddress as we are, the path must not have a preceeding slash
+            response = _client.GetAsync(path + "/" + param).Result;
+            if (response.IsSuccessStatusCode)
+            {
+                content = response.Content;
+            }
+            else
+            {
+                string errorMessage = "Response: " + response.Content.ReadAsStringAsync().Result + "\nAPI path: " + this._client.BaseAddress.ToString() + path + "/" + param;
+                if (response.StatusCode.ToString() == notFound)
+                {
+                    // If trial is not found, log 404 message and return content as null
+                    log.Debug(errorMessage);
+                }
+                else
+                {
+                    // If response is other error message, log and throw exception
+                    log.Error(errorMessage);
+                    throw new Exception(errorMessage);
+                }
+            }
+
+            return content;
+        }
+
+        public static async Task<GlossaryTerm> ReadAsJsonAsync<GlossaryTerm>(HttpContent content)
+        {
+            string json = await content.ReadAsStringAsync();
+            GlossaryTerm result = JsonConvert.DeserializeObject<GlossaryTerm>(json);
+            return result;
+        }
```

Comment:
This method is only referenced in `GlossaryAPIClient`. Unless there is an intended unit test, it should be either `private` or `protected`.